client/app/scripts/liveblog-themes/module.js

```diff
@@ -193,6 +158,66 @@
             //return to blog list page
             $location.path('/liveblog/');
         };
+
+        function loadThemes() {
+            // load only global preference for themes.
+            api.global_preferences.query({'where': {'key': 'theme'}}).then(function(global_preferences) {
+                $scope.globalTheme = _.find(global_preferences._items, function(item) {
+                    return item.key === 'theme';
+                });
+            });
+            // load all the themes.
+            // TODO: Pagination
+            return api.themes.query({timestamp: Date()}).then(function(data) {
+                var themes = data._items;
+                themes.forEach(function(theme) {
+                    // create criteria to load blogs with the theme.
+                    var criteria = {
+                            source: {
+                                query: {filtered: {filter: {term: {'theme._id': theme._id}}}}
+                            }
+                        };
+                    api.blogs.query(criteria).then(function(data) {
+                        theme.blogs_count = data._meta.total;
+                        // TODO: Pagination. Will only show the first results page
+                        theme.blogs = data._items;
+                    });
+                    parseTheme(theme);
+                });
+                // object that represent the themes hierachy
+                var themes_hierachy = getHierachyFromThemesCollection(themes);
+                // update the scope
+                angular.extend($scope, {
+                    themesHierachy: themes_hierachy,
+                    themes: themes,
+                    loading: false
+                });
+                return themes;
+            });
+        }
+
+        $scope.uploadZippedTheme = function(e) {
```

Comment:
the name is not relevant. it should be plain upload Theme
It is really up to the endpoint to establish the compression.